### PR TITLE
Customize logs table

### PR DESCRIPTION
**Pull request by SpennyNDaJets**
## Summary
Allow users to select the columns they want displayed on the traces table. Save this data in local storage, so they will see their changes in other sessions.

## How did you test this change?
1) Load the logs table
- [ ] Columns and data is loaded properly
2) Click the 3 dots on the far right column and add/remove columns
- [ ] Columns and data responds appropriately
3) Reload the page
- [ ] Columns are persistent
4) Load the page in a new browser/incognito window
- [ ] Default columns are displayed
5) View the logs on the traces or session player
- [ ] Columns are not customizable
- [ ] Default columns are shown
- [ ] No effect to logs table

https://www.loom.com/share/fee311ef2b8247f9ad669ee4cfb734c5?sid=9f2623ae-18c0-4db5-ac04-6e6b89453205

## Are there any deployment considerations?
N/A

## Does this work require review from our design team?
Sure

### CODE REVIEW

File: frontend/src/pages/Traces/TracesList.tsx [hunk 1]
```diff
@@ -311,7 +319,7 @@ const TracesTableRow = React.memo<TracesTableRowProps>(
 		return (
 			prevProps.virtualRowKey === nextProps.virtualRowKey &&
 			prevProps.isSelected === nextProps.isSelected &&
-			prevProps.gridColumns === nextProps.gridColumns
+			isEqual(prevProps.gridColumns, nextProps.gridColumns)
```
Comment (SpennyNDaJets) [inline frontend/src/pages/Traces/TracesList.tsx:322]: Updated this to do deep equal check and prevent re-renders do to save by reference differences

**Review by ccschmitz — APPROVED** [review]
Looks great overall! One thing I was wondering about was the labels in the dropdown menu appear to all be in snake case, is that correct, or maybe something I don't have configured correctly in dev on my end?

Screenshot 2024-01-31 at 6 43 37 AM

Also noticed some jank in the rendering of the table when I added a few columns with longer names. Probably not something to worry about right now, but thought I'd mention it.

Screenshot 2024-01-31 at 6 44 26 AM

File: backend/clickhouse/logs.go [hunk 2]
```diff
@@ -161,6 +163,7 @@ func (client *Client) ReadLogs(ctx context.Context, projectID int, params modelI
 				ServiceName:     &result.ServiceName,
 				ServiceVersion:  &result.ServiceVersion,
 				Environment:     &result.Environment,
+				ProjectID:       int(result.ProjectId),
```
[the remark this answers is not in the document]
Comment (SpennyNDaJets) [reply backend/clickhouse/logs.go:166]: Like you said, this is only being used for the link generation on the client. The alternative is the grab the project id out from the url, and pass it into the render.

I chose this way to:
- keep consistent with the traces, which includes the `ProjectID`
- Offer flexibility in the future, if we end up showing a table on a non-project id related page

But don't feel too strongly about this at the current moment

[the remark this answers is not in the document]
Comment (ccschmitz) [reply backend/clickhouse/logs.go:166]: Sorry, we talked about this when pairing but I forgot to delete the comment before submitting review - my bad!

**Comment by SpennyNDaJets** [comment]
> Looks great overall! One thing I was wondering about was the labels in the dropdown menu appear to all be in snake case, is that correct, or maybe something I don't have configured correctly in dev on my end?

Sort of correct, we are just showing the keys from the key search, which is how the user would send us any of the data in the logs/traces. I originally tried showing a label-like text here, but discussed this with Zane, and opted to show the keys as they are to avoid confusing the user.

There could be a little confusion to clicking a key, and seeing a label-like text appear in the table, but this should be easy to link, and the user will be able to rename the label in https://github.com/highlight/highlight/pull/7618

**Comment by SpennyNDaJets** [comment]
> Also noticed some jank in the rendering of the table when I added a few columns with longer names. Probably not something to worry about right now, but thought I'd mention it.

Thanks for raising! I will address in https://github.com/highlight/highlight/pull/7618

File: backend/clickhouse/logs.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (SpennyNDaJets) [reply backend/clickhouse/logs.go:166]: Not at all! Good to get our thoughts down here!

**Pull request merged**